prost-build/src/code_generator.rs

```diff
@@ -251,26 +251,35 @@ impl<'a> CodeGenerator<'a> {
 
     fn append_type_attributes(&mut self, fq_message_name: &str) {
         assert_eq!(b'.', fq_message_name.as_bytes()[0]);
-        // TODO: this clone is dirty, but expedious.
-        if let Some(attributes) = self.config.type_attributes.get(fq_message_name).cloned() {
-            self.push_indent();
-            self.buf.push_str(&attributes);
-            self.buf.push('\n');
+        let mut buf = String::new();
+        for attribute in self.config.type_attributes.get(fq_message_name) {
+            for _ in 0..self.depth {
+                buf.push_str("    ");
+            }
+            buf.push_str(&attribute);
+            buf.push('\n');
+        }
+        if !buf.is_empty() {
+            self.buf.push_str(&buf);
         }
     }
 
     fn append_field_attributes(&mut self, fq_message_name: &str, field_name: &str) {
         assert_eq!(b'.', fq_message_name.as_bytes()[0]);
-        // TODO: this clone is dirty, but expedious.
-        if let Some(attributes) = self
+        let mut buf = String::new();
+        for attribute in self
             .config
             .field_attributes
             .get_field(fq_message_name, field_name)
-            .cloned()
         {
-            self.push_indent();
-            self.buf.push_str(&attributes);
-            self.buf.push('\n');
+            for _ in 0..self.depth {
+                buf.push_str("    ");
+            }
+            buf.push_str(&attribute);
+            buf.push('\n');
+        }
+        if !buf.is_empty() {
```

Comment:
Is this check needed? Appending a empty str should be a noop

Reply:
Probably not

Reply:
Removed